packages/components/src/Banner/Banner.tsx

```diff
@@ -138,3 +113,27 @@ export default function Banner({
     </article>
   );
 }
+
+/**
+ * This should import a Heading element type
+ */
+Banner.Title = (props: { children?: React.ReactNode; as: HeadingElement }) => {
+  return props.children ? (
+    <Heading as={props.as} color={"inherit"} size="h3">
+      {props.children}
+    </Heading>
+  ) : null;
+};
+Banner.Title.displayName = "Banner.Title";
```

Comment:
I played around a little bit locally and the root issue is `displayName` is only defined on component types. I don't understand why TS insists on returning type `Element`, when `Heading` itself is correctly typed as a component though 😭 

Two ways I found of working around it are:
1) Casting as `React.FC` (aka `FunctionalComponent` type):
```
type TitleProps = {
  children?: React.ReactNode;
  as: HeadingElement;
};
/**
 * This should import a Heading element type
 */
const BannerTitle: React.FC<TitleProps> = (
  props: TitleProps,
): React.ReactElement | null => {
  return props.children ? (
    <Heading as={props.as} color={"inherit"} size="h3">
      {props.children}
    </Heading>
  ) : null;
};
Banner.Title = BannerTitle;
Banner.Title.displayName = "Banner.Title";
```
Unfortunately, it's extra verbose because I couldn't get directly casting `Banner.Title: React.FC<TitleProps>` to work

2) Just silencing the TS error 😬 
```suggestion
// @ts-expect-error TS thinks this is not a component type but it is 
Banner.Title.displayName = "Banner.Title";
```

or 3) as you suggested, updating the docstring (also don't know why the docstring would disappear??)

Reply:
for some reason, setting
```
Banner.displayName = "Banner";
```
_before_ setting the other displayNames gets the docstring to show up for me 🧐

Reply:
I updated the PR to cast the types to `React.FC` – thank you so much for figuring that out @anniehu4 ❤️ ❤️ ❤️